src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsReconciliationService.kt

```diff
@@ -49,21 +53,26 @@ class ContactPersonProfileDetailsReconciliationService(
 
   suspend fun generateReconciliationReport(activePrisonersCount: Long): Mismatches = if (reconciliationTurnedOn) {
     coroutineScope {
-      mismatchesChannel().also { mismatches ->
-        producePrisonerIds(activePrisonersCount)
-          .also { ids -> checkPrisoners(ids, mismatches) }
-      }.waitForMismatches()
+      val idProducer = producePrisonerIds(activePrisonersCount)
+      val mismatchConsumer = Channel<String>(capacity = UNLIMITED)
+      (1..parallelJobs)
+        .map { launchCheckPrisoners(idProducer, mismatchConsumer) }
+        .let { jobs -> mismatchConsumer.waitForMismatches(jobs) }
```

Comment:
Allows us to set the maximum number of prisoner checks running in parallel. This gives us control over the load we place on NOMIS.